test/mapmatch.cc

```diff
@@ -150,6 +150,56 @@ std::string json_escape(const std::string& unescaped) {
   return escaped;
 }
 
+std::string output_shape(const valhalla::Api& api) {
```

Comment:
Our tests so far only exercise the end-to-end results, but we lack unit tests for (approximately any of the meili-related functions changing in this PR. Do we intend on adding them?

/cc @kevinkreiser @yuzheyan

Reply:
while i do think its a good idea overall, i dont think its a good idea at this time. we can always add finer detail unit tests over time. at the moment i think the map matching unit tests do have quite good code coverage (as they are probably the largest test in the repository). i do think though that you loose the detail of what you are exorcising a bit if you dont spell it out in your tests. so like we have all those namespace functions in meili, and a lot of unit tests here are designed to exorcise them but you cant tell that if you dont know the code. so yeah i do think those fine grain tests would help for readibility but i dont think they should block this PR